[Joy Foo] ip

yezkez10 opened the pull request:
# ToDoBot 📃

> "You put the work in and then that confidence builds up and you start to do things you really didn't know you can do."
>
> -[James Harden](https://www.google.com/search?sca_esv=562741715&rlz=1C1CHZN_enSG1047SG1047&sxsrf=AB5stBhKYQgnAOVs5-ECKwxh9O1Wb992MA:1693920082402&q=motivational+quotes+athletes&tbm=isch&source=lnms&sa=X&sqi=2&ved=2ahUKEwig-aeGyJOBAxVm2TgGHUnrBXsQ0pQJegQIDhAB&biw=1024&bih=513&dpr=1.88#imgrc=3YTPWsMci9VbFM) (some NBA player)

ToDoBot shows you that a problem written down is a prolem half-solved. It's
- text-activated
- easy to use
- ~~pretty~~ _SUPA_ cool

## Features
- [ ] Add
- [ ] Delete
- [X] Mark
- [ ] Unmark

# How to download
1. click [here](https://github.com/yezkez10/ip).
2. download
3. use it!

Here's a snippet!
```
 public Duke(String filePath) {
        ui = new Ui();
        storage = new Storage(filePath);
        tasks = new TaskList(storage.getTaskList());
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0 to 2 cos i forgot.. tehe

Commit:
Debug level-3

Commit:
Do Level-3. why did it take me 5 hours?? But finally clutched it

Commit:
Do level-4. Abstraction heart shape heart shape

Commit:
Do Level-5

Commit:
Complete Level-6

Commit:
Testing

Commit:
Testing

Commit:
Level-7

Commit:
Edit Deadline to deal with dates

Commit:
Debug level-8. Still can't parse when no time involved.

Commit:
Redo Level-4

Commit:
Redo Level-6 Delete

Commit:
Redo Level-7: Save

Commit:
Redo Level-8

Commit:
Add load task function

Commit:
Add handling of no-such-file case

Commit:
Add OOP

Commit:
Organise files into packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add non-trivial JUnit tests

Commit:
Add JavaDoc comments

Commit:
Level-9

chonguschonguschongus commented on src/main/java/duke/exceptions/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exceptions;
```
Exceptional exception handling! Joy! 👍

chonguschonguschongus commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+package duke.tasks;
```
Great extracting of the Task class!

chonguschonguschongus commented on src/main/java/duke/tools/KeywordEnum.java:
```diff
@@ -0,0 +1,17 @@
+package duke.tools;
```
Nice enumeration! 🤟

chonguschonguschongus reviewed (commented):
LGTM. Very clean code! Good work Joy!

> "Some people want it to happen, some wish it would happen, others make it happen" -Michael Jordan

yezkez10 committed:
Update README.md

Commit:
edit build.gradle

josepholim commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,60 @@
+        String restOfFrom = start.substring(from.length()).trim();
+
+        String end = descrArr[2];
+        String[] parts2 = end.split(" ");
```
Can the variable name be more explanatory? Instead of `parts2`? 🤔

josepholim commented on src/main/java/duke/tools/KeywordEnum.java:
```diff
@@ -0,0 +1,17 @@
+
+    public static KeywordEnum assign(String task) {
+        if (task.startsWith("list")) return KeywordEnum.LIST;
+        else if (task.startsWith("bye")) return KeywordEnum.BYE;
```
Try to use Egyptian Style brackets for more readability :D

josepholim commented on src/main/java/duke/tools/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+
+        while (true) {
+            switch(keywordEnum) {
+                case LIST:
```
Iirc the coding style used in this course suggests for no indentation for `case` clauses 🤔

josepholim commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+        return res;
+    }
+
+    public String writtenFormat() {
```
Verbs should be used for functions/methods naming. Probably `getWrittenFormat` can be a better name for this method? 🤔

josepholim reviewed (commented):
LGTM 👍 :D

Commit:
Add Level-10 GUI

Commit:
Add GUI

Commit:
Merge branch 'master' of https://github.com/yezkez10/ip

Commit:
Add assertions.
Assertions define assumptions for runtime to verify.

Assertions used liberally for error handling.

Assertions used to verify assumptions before runtime catches error from assumption.

yezkez10 committed:
Merge pull request #1 from yezkez10/branch-A-Assertions

Add assertions.

Commit:
Improve code quality.

Production code needs to be of high quality.

Many methods have been refactored to return Strings,
so JavaDoc has been updated to reflect these changes.

Tried to SLAP hard most importantly.

Commit:
MainWindow.java initialise and handleUserInput

initialise does not print welcome message, and handleUserInput immediately exits app once 'bye' is inputted

Printing a welcome message before handling inputs makes it more presentable,
and waiting 5 secs before exiting the app makes the app feel closer to a complete production.

Others:
Shortened some methods in Duke.java to improve code quality.

Commit:
Refactor code
Methods were not abstracted well, a lot of unnecessary lines of code that could be truncated.
Added parsers to SLAP better, also separated date reformatting from date validity check.
Done this way so that could SLAP better.

Commit:
Improve code quality

For code to be more readable.

Deleting dead code and adding javadoc improves code quality.

yezkez10 committed:
Update README.md

Commit:
Finalise features

Commit:
Merge branch 'master' of https://github.com/yezkez10/ip

yezkez10 committed:
Update README.md

yezkez10 committed:
Update README.md

Commit:
Add Ui.png

Commit:
Merge branch 'master' of https://github.com/yezkez10/ip

yezkez10 committed:
Update README.md, add Ui.png

yezkez10 committed:
Update README.md formatting

Commit:
Fix issues with relative path. Update tests
